app/controllers/api/v1/trainers/pokemons_controller.rb

```diff
@@ -1,4 +1,26 @@
 class Api::V1::Trainers::PokemonsController < ApplicationController
+  def create
+    pokemon_name = pokemon_params[:name]
+    trainer_id = params[:trainer_id]
+    begin
+      pokemon = Pokemon.populate_with_api_data(pokemon_params, pokemon_name, trainer_id)
+      if pokemon.save
+        render json: PokemonSerializer.new(pokemon), status: :created
+      else
+        render json: ErrorSerializer.format_error(pokemon.errors, 422), status: :unprocessable_entity
+      end
+    rescue StandardError => e
```

Comment:
Looks good, one thing we should discuss from a stylistic/consistency perspective is is we want to write "exception" the long way or shorthand as "e".